src/rpcblockchain.cpp

```diff
@@ -1063,1370 +1097,2389 @@ int64_t AmountFromDouble(double dAmount)
-                            dTotalToSend += dAmount;
-                            int64_t nAmount = AmountFromDouble(dAmount);
-                            CScript scriptPubKey;
-                            scriptPubKey.SetDestination(address.Get());
-                            totalAmount += nAmount;
-                            vecSend.push_back(make_pair(scriptPubKey, nAmount));
-                        }
-                    }
+                    CBitcoinAddress address(sAddress);
+                    if (!address.IsValid())
+                        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, string("Invalid Gridcoin address: ")+sAddress);
+
+                    if (setAddress.count(address))
+                        throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, duplicated address: ")+sAddress);
+
+                    setAddress.insert(address);
+                    dTotalToSend += dAmount;
+                    int64_t nAmount = AmountFromDouble(dAmount);
+                    CScript scriptPubKey;
+                    scriptPubKey.SetDestination(address.Get());
+                    totalAmount += nAmount;
+                    vecSend.push_back(make_pair(scriptPubKey, nAmount));
                 }
             }
-
-            EnsureWalletIsUnlocked();
-            // Check funds
-            double dBalance = GetTotalBalance();
-            if (dTotalToSend > dBalance)
-                throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Account has insufficient funds");
-            // Send
-            CReserveKey keyChange(pwalletMain);
-            int64_t nFeeRequired = 0;
-            bool fCreated = pwalletMain->CreateTransaction(vecSend, wtx, keyChange, nFeeRequired);
-            LogPrintf("Transaction Created.");
-            if (!fCreated)
-            {
-                if (totalAmount + nFeeRequired > pwalletMain->GetBalance())
-                    throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Insufficient funds");
-                throw JSONRPCError(RPC_WALLET_ERROR, "Transaction creation failed");
-            }
-            LogPrintf("Committing.");
-            // Rain the recipients
-            if (!pwalletMain->CommitTransaction(wtx, keyChange))
-            {
-                LogPrintf("Commit failed.");
-                throw JSONRPCError(RPC_WALLET_ERROR, "Transaction commit failed");
-            }
-            std::string sNarr = "Rain successful:  Sent " + wtx.GetHash().GetHex() + ".";
-            LogPrintf("Success %s",sNarr);
-            entry.push_back(Pair("Response", sNarr));
-            results.push_back(entry);
         }
     }
-    else if (sItem == "neuralrequest")
-    {
 
-            std::string response = NeuralRequest("REQUEST");
-            entry.push_back(Pair("Response", response));
-            results.push_back(entry);
+    EnsureWalletIsUnlocked();
+    // Check funds
+    double dBalance = GetTotalBalance();
 
+    if (dTotalToSend > dBalance)
```

Comment:
Should this also check if dTotalToSend does not exceed dBalance + txfee?

Reply:
fee gets checked below